### PR TITLE
New Recipe: gat v0.19.1

### PR DESCRIPTION
**Pull request by terasakisatoshi**
This PR introduces `gat`  (cat alternative written in Go). https://github.com/koki-develop/gat

Permission to use this code has been granted by the author. 
See https://github.com/koki-develop/gat/issues/86 for further information.

The recipe `G/gat/build_tarballs.jl` is based on [H/HelloWorldGo/build_tarballs.jl](https://github.com/JuliaPackaging/Yggdrasil/blob/master/H/HelloWorldGo/build_tarballs.jl)

When PR is merged https://github.com/terasakisatoshi/TerminalGat.jl can be registered in the Julia's General registry.

### CODE REVIEW

File: G/gat/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/koki-develop/gat.git", "173737297556cbcdf6c1e3145ae8baa1c4a68bdf")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/gat
+go build -o ${prefix}/bin/gat${exeext}
```
Comment (giordano) [inline G/gat/build_tarballs.jl:14]:
```suggestion
go build -o "${bindir}/gat${exeext}"
```

File: G/gat/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+products = [
+    ExecutableProduct("gat", :gat)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", compilers = [:go, :c])
```
Comment (giordano) [inline G/gat/build_tarballs.jl:31]: Missing newline
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", compilers = [:go, :c])

```

**Comment by giordano** [comment]
Thanks for your contribution!

**Comment by terasakisatoshi** [comment]
Thank you for your review!

**Pull request merged**